### PR TITLE
Free various struct membuffer in different functions

### PR DESCRIPTION
**Pull request by sfuchs79**
### Describe the pull request:

- [x] Bug fix
- [ ] Functional change
- [ ] New feature
- [x] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

@neolit123 correctly highlighted that I introduced a struct membuffer in my code and forgot to free it.
While learning how to fix it correctly I think I found some more missing free_buffer().

But please check this carefully because maybe I misunderstood s.th.!

### Mentions:

@neolit123

### CODE REVIEW

File: core/save-git.c [hunk 1]
```diff
@@ -596,6 +596,7 @@ static int save_one_divecomputer(git_repository *repo, struct dir *tree, struct
 	ret = blob_insert(repo, tree, &buf, "Divecomputer%c%03u", idx ? '-' : 0, idx);
 	if (ret)
 		report_error("divecomputer tree insert failed");
+	free_buffer(&buf);
```
Comment (neolit123) [inline core/save-git.c]: this is a double `free_buffer()`. look at `blob_insert()` - it already calls `free_buffer()`.
it's not ideal that the ownership is transferred to `blob_insert()`, too.

File: core/linux.c [hunk 2]
```diff
@@ -42,6 +42,7 @@ void subsurface_user_info(struct user_info *user)
 		gethostname(hostname, sizeof(hostname));
 		put_format(&mb, "%s@%s", username, hostname);
 		user->email = mb_cstring(&mb);
+		free_buffer(&mb);
```
Comment (neolit123) [inline core/linux.c:45]: this one looks good.

File: core/save-git.c [hunk 3]
```diff
@@ -632,8 +633,10 @@ static int save_one_picture(git_repository *repo, struct dir *dir, struct pictur
 		put_format(&namebuf, "PIC-%s", pic->hash);
 		error = blob_insert_fromdisk(repo, dir, localfn, mb_cstring(&namebuf));
 		free((void *)localfn);
+		free_buffer(&namebuf);
 	}
 #endif
+	free_buffer(&buf);
```
Comment (neolit123) [inline core/save-git.c]: double `free_buffer()`. there is a `blob_insert()` call above.

File: core/save-git.c [hunk 4]
```diff
@@ -895,6 +898,7 @@ static void save_settings(git_repository *repo, struct dir *tree)
 		put_string(&b, "prefs PO2_GRAPH\n");
 
 	blob_insert(repo, tree, &b, "00-Subsurface");
+	free_buffer(&b);
```
Comment (neolit123) [inline core/save-git.c]: double `free_buffer()`. there is a `blob_insert()` call above.

File: core/save-git.c [hunk 5]
```diff
@@ -903,6 +907,7 @@ static void save_divesites(git_repository *repo, struct dir *tree)
 	struct membuffer dirname = { 0 };
 	put_format(&dirname, "01-Divesites");
 	subdir = new_directory(repo, tree, &dirname);
+	free_buffer(&dirname);
```
Comment (neolit123) [inline core/save-git.c:906]: this one looks good.

File: core/save-git.c [hunk 6]
```diff
@@ -946,6 +951,8 @@ static void save_divesites(git_repository *repo, struct dir *tree)
 			}
 		}
 		blob_insert(repo, subdir, &b, mb_cstring(&site_file_name));
+		free_buffer(&b);
```
Comment (neolit123) [inline core/save-git.c]: double `free_buffer()`. there is a `blob_insert()` call above.

File: core/save-git.c [hunk 7]
```diff
@@ -946,6 +951,8 @@ static void save_divesites(git_repository *repo, struct dir *tree)
...
+		free_buffer(&site_file_name);
```
Comment (neolit123) [inline core/save-git.c]: this one looks good.

File: desktop-widgets/subsurfacewebservices.cpp [hunk 8]
```diff
@@ -126,6 +126,7 @@ bool DivelogsDeWebServices::prepare_dives_for_divelogs(const QString &tempfile,
 			goto error_close_zip;
 		}
 		free((void *)membuf);
+		free_buffer(&mb);
```
Comment (neolit123) [inline desktop-widgets/subsurfacewebservices.cpp]: the `mb` variable itself is allocated on the stack.
the above call already handles the memory de-allocation - `free((void *)membuf);`
but i would remove the `free((void *)membuf);` line and leave the line you've added `free_buffer(&mb);`

File: profile-widget/diveeventitem.cpp [hunk 9]
```diff
@@ -186,6 +186,7 @@ void DiveEventItem::setupToolTipString(struct gasmix *lastgasmix)
 			name += QString::fromUtf8(mb.buffer, mb.len);
 		}
 		*lastgasmix = *mix;
+		free_buffer(&mb);
```
Comment (neolit123) [inline profile-widget/diveeventitem.cpp]: this is a nitpick.
i would reduce the scope of the `free_buffer(&mb);` here and place it bellow the `name += QString::fromUtf8(mb.buffer, mb.len);` line. it is relevant only if the `put_format()` call above succeeds.

**Review by neolit123 — CHANGES_REQUESTED** [review]
thanks for the PR, actually some of the changes can cause some trouble, thus i've added a review for these. requesting a review from others too - @mturkia @dirkhh

**Comment by sfuchs79** [comment]
Updates done. Please review again!

**Review by neolit123 — APPROVED** [review]
lgtm, thanks. someone needs to double check.

**Comment by dirkhh** [comment]
I don't think my brain is clear enough to do a valid code review. Hopefully I'll feel better by Monday. This flu sucks...

**Comment by dirkhh** [comment]
ok, lgtm

**Pull request merged**